simple_app/app_simple.py

```diff
@@ -171,6 +187,141 @@ def autocomplete():
     return jsonify(alljsonlist=all_results)  # wraps all of the object names as a list, into a .json for server use
 
 
+@app_simple.route('/write/<key>', methods=['GET', 'POST'])
+def create_file_for_download(key: str):
+    """
+    Creates and downloads the shown dataframe from solo results
+
+    Parameters
+    ----------
+    key: str
+        The dataframe string
+    """
+    query = curdoc().template_variables['query']
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    resultdict: dict = db.inventory(query)  # get everything about that object
+    everything = Inventory(resultdict, args, rtnmk=False)
+    if key in resultdict:
+        results: pd.DataFrame = getattr(everything, key.lower())
+        fname = write_file(results).split('/')[-1]
+        return redirect(url_for('download_file', filename=fname))
```

Comment:
Why do we have to generate a file first to send it to the user? Can we not send the data directly? Something along the lines of what is described in https://flask.palletsprojects.com/en/1.0.x/patterns/streaming/